weather.py

```diff
@@ -1,109 +1,134 @@
-    data.update(('wind_' + attr, wind.attrs[attr]) for attr in wind.attrs)
-    data['wind_direction_fuzzy'] = fuzzy_direction(int(data['wind_direction']))
-    data['wind_direction_arrow'] = arrow_direction(int(data['wind_direction']))
-    # Atmospheric conditions - humidity, visibility, pressure
-    data.update(s.find('yweather:atmosphere').attrs)
-    # Astronomical conditions - sunrise / sunset
-    data.update(fix_sunrise_sunset(s.find('yweather:astronomy').attrs))
-    return format.format(**data)
-
-sunrise_sunset_re = re.compile(r'(?P<hour>\d+):(?P<minute>\d+) (?P<am_pm>am|pm)')
-def fix_sunrise_sunset(attrs):
-    for key, value in attrs.items():
-        match = sunrise_sunset_re.fullmatch(value)
-        if match:
-            yield key, '{hour}:{minute:>02} {am_pm}'.format(**match.groupdict())
-        else:
-            yield key, value
+    data['unit_pressure'] = 'hPa'
+    if unit_temp == 'fahrenheit':
+        data['unit_temperature'] = 'F'
+    else:
+        data['unit_temperature'] = 'C'
+    if unit_speed == 'miles_hour':
+        data['unit_speed'] = 'mph'
+    else:
+        data['unit_speed'] = 'm/s'
+
+    loc = obs.get_location()
+    weather = obs.get_weather()
+
+    data['city'] = loc.get_name()
+    data['country'] = loc.get_country()
+    data['temp'] = round(weather.get_temperature(unit=unit_temp)['temp'])
+    data['text'] = weather.get_detailed_status()
+    data['humidity'] = weather.get_humidity()
+    data['pressure'] = weather.get_pressure()['press']
+
+    wind = weather.get_wind(unit=unit_speed)
+    # Wind direction is sometimes unset; I'm assuming this occurs when
+    # different weather stations for the same location report the wind
+    # blowing in conflicting directions
+    if 'deg' not in wind:
+        wind['deg'] = None
+    data['wind_speed'] = round(wind['speed'])
+    data['wind_direction'] = wind['deg']
+    data['wind_direction_fuzzy'] = fuzzy_direction(wind['deg'])
+    data['wind_direction_arrow'] = arrow_direction(wind['deg'])
+
+    data['sunrise'] = unix_to_hhmm(weather.get_sunrise_time())
+    data['sunset'] = unix_to_hhmm(weather.get_sunset_time())
+    return format_str.format(**data)
+
+def count_set(*args):
+    count = 0
+    for a in args:
+        if a:
+            count += 1
+    return count
 
 if __name__ == '__main__':
     p = argparse.ArgumentParser()
-    p.add_argument('woeid')
     p.add_argument('--format', metavar='F',
-                   default='{city}, {region}: {text}, {temp}\u00b0{unit_temperature}',
+                   default='{city}, {country}: {text}, '
+                           '{temp}\u00b0{unit_temperature}',
                    help="format string for output")
     p.add_argument('--position', metavar='P', type=int, default=-2,
                    help="position of output in JSON when wrapping i3status")
-    p.add_argument('--unit', metavar='U', default='f',
+    p.add_argument('--unit-temperature', metavar='U', default='fahrenheit',
```

Comment:
If you don't go with the above suggestion to switch back to F/C here, metavar should probably be a word since the choices are now words rather than characters.
```suggestion
    p.add_argument('--unit-temperature', metavar='UNIT', default='fahrenheit',
```